evm/src/arithmetic/mul.rs

```diff
@@ -10,23 +10,49 @@
 //!
 //!    A = \sum_{i=0}^15 a[i] β^i,
 //!
-//! where β = 2^16. To verify that A, B and C satisfy the equation we
-//! proceed as follows. Define a(x) = \sum_{i=0}^15 a[i] x^i (so A = a(β))
-//! and similarly for b(x) and c(x). Then A*B = C (mod 2^256) if and only
-//! if there exist polynomials q and m such that
+//! where β = 2^16 = 2^LIMB_BITS. To verify that A, B and C satisfy
+//! the equation we proceed as follows. Define
 //!
-//!    a(x)*b(x) - c(x) - m(x)*x^16 - (β - x)*q(x) == 0.
+//!    a(x) = \sum_{i=0}^15 a[i] x^i
+//!
+//! (so A = a(β)) and similarly for b(x) and c(x). Then A*B = C (mod
+//! 2^256) if and only if there exists q such that the polynomial
+//!
+//!    a(x) * b(x) - c(x) - m(x) * q(x)
```

Comment:
You can make the rest of this documentation clearer by stating upfront that `m` has a concrete value (which is 2**256) and thus `m(x)` is (dot dot dot).